### PR TITLE
`Assessment`: Fix sorting in exercise scores table

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

Fix sorting in exercise scores table and make the sorting function more resilient for the future.

### Description

The sorting function was not able to handle potential additions of optional chaining and array accesses using `[0]` in the string. This is now handled correctly, fixing the sorting of some columns in the exercise scores table.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise with some submissions

1. Log in to Artemis
2. Navigate to the exercise scores table
3. Check that sorting each column works as expected

#### Performance Review
- [x] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by SolizerCodes — APPROVED** [review]
Sorting all the different fields works on ts5

**Review by DominikRemo — CHANGES_REQUESTED** [review]
Tested on ts2 and it works well.

However, your Client Tests fail to compile:
`Error: src/test/javascript/spec/service/sort.service.spec.ts(46,9): error TS2322: Type 'undefined' is not assignable to type '{ i: number; }[]'.`

**Review by DominikRemo — APPROVED** [review]
Thank you for fixing the client test

File: src/main/webapp/app/shared/service/sort.service.ts [hunk 1]
```diff
@@ -95,7 +95,9 @@ export class SortService {
     }
 
     private static customGet(object: any, path: string, defaultValue: any) {
-        const pathArray = path.split('.').filter((key) => key);
+        // Get rid of all optional chainings as they are handled down below. After that split the path into all array and attribute accesses
```
Comment (b-fein) [inline src/main/webapp/app/shared/service/sort.service.ts:98]:
```suggestion
        // Get rid of all optional chainings as they are handled down below. After that split the path into all array and attribute accesses
        // Example: path 'some?.path[0].x' will be split into pathArray ['some', 'path', '0', 'x']
```
Small example makes it easy to understand what the overall intention of the replacements and splitting is.

**Review by b-fein — APPROVED** [review]

**Review by terlan98 — DISMISSED** [review]
Tested on TS2. Works as expected 👍 
Code LGTM

**Review by b-fein — APPROVED** [review]
Code looks good

**Review by terlan98 — APPROVED** [review]
Reapproved

**Review by sven0311 — APPROVED** [review]
Tested on Ts1. Sorting works as expected

**Review by DominikRemo — APPROVED** [review]
Reapproved

**Review by SolizerCodes — APPROVED** [review]
Reapprove

**Review by undernagruzez — APPROVED** [review]
Works as expected, LGTM

**Pull request merged**